### DISCUSSION TITLE
general recommendations on HPC [help]

### DISCUSSION THREAD

**Discussion by rcorty** [ropensci/targets#1060] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I understand there are primarily two options for using {targets} on high-performance compute clusters: {future} and {clustermq}.  I read over the manual and it seems like the main difference between the two is whether they use persistent or transient workers. My question is this -- if I'm not particularly concerned with "compute efficiency" and more concerned with "developer efficiency", which one should I generally choose? I see that {clustermq} is not yet at version 1 and {future} is, but {future.batchtools} (which I believe I would need) is not.  I also see that {crew} is coming soon but my impression from the current version number is that it's early stage.

In short, what's the simplest way to get my targets to run on my university's SLURM cluster? If that question is not answerable, what are the facts that would be needed to answer, e.g. typical wait-time, typical target run time, typical # of targets per projects?

**Comment by wlandau** [comment, marked as answer]
`clustermq` is generally faster for a large number of quick-running targets, and `future` is sometimes better for long-running targets if your university has strict walltime policies. Other than that, depending on your toolchain, `clustermq` may be hard to install. I think those are the main factors.

I do plan to write a SLURM launcher for [`crew.cluster`](https://github.com/wlandau/crew.cluster), which could outperform `clustermq` in data transfer speed and will offer a smooth continuum between persistent and transient workers.
